Inbox slidein

Crabcyborg opened the pull request:
Fixes https://github.com/Strategy11/formidable-pro/issues/4862
Requires https://github.com/Strategy11/formidable-pro/pull/4872 for images to appear.

**TODO**
- [x] I don't believe we are allowed to include images by URL according to the WordPress guidelines.
- [x] Add the close icon to dismiss the dialog and dismiss the inbox message.
- [x] Make sure to change the UTM attributes appropriately for slidein and possibly banner if it is not already sending a UTM for banner.

<img width="503" alt="Screen Shot 2024-03-21 at 11 49 43 AM" src="https://github.com/Strategy11/formidable-forms/assets/9134515/82a0bce9-d30e-49b0-95f7-64d9e4682bcc">

The dismiss icon in the review request has also been updated.
<img width="476" alt="Screen Shot 2024-03-21 at 11 50 09 AM" src="https://github.com/Strategy11/formidable-forms/assets/9134515/26c2135d-ea05-478b-b29a-1e5cb4070398">


**For testing**

You can filter the inbox option and fake messages. This is what I have right now:
```
add_filter( 'option_frm_inbox', 'fake_inbox' );

function fake_inbox( $settings ) {
	return array(
		'slidein_test' => array(
			'message' => '🎉 Exciting Announcement! 🎉 Don\'t miss out on our amazing new feature! Now you can do more than ever before!',
			'subject' => 'New features &amp; more!',
			'icon'    => '',
			'cta'     => '<a href="https://formidableforms.com/knowledgebase/latest-videos/?utm_source=WordPress&utm_medium=inbox&utm_campaign=liteplugin&utm_content=videos" class="button-primary frm-button-primary" target="_blank" rel="noopener noreferrer">Give it a try!</a>',
			'image'   => 'https://qa.formidableforms.com/mike2/wp-content/uploads/sites/17/formidable/501/Screen-Shot-2024-03-11-at-4.13.16-PM.png',
			'expires' => '',
			'who'     => array( 'all' ),
			'type'    => 'test',
			'key'     => 'slidein_test',
			'starts'  => '',
			'created' => time(),
			'slidein' => 'Don\'t miss out on our amazing new feature! Now you can do more than ever before!',
		)
	);
}
```

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
    - Introduced inbox slide-in functionality for enhanced user notifications.
    - Improved handling of banner and slide-in messages, including new methods for message management.
    - Added new styling for dismissible elements and inbox slide-in feature, improving visual appeal and user interaction.
- **Bug Fixes**
    - Refined the conditions for showing floating links for better user experience.
    - Updated the HTML structure and event handling for dismissing review notices, ensuring smoother user interactions.
- **Refactor**
    - Enhanced code readability and maintainability in JavaScript files by using modern variable declarations (`const` and `let`).
    - Streamlined message read, dismiss, and expiration checks for efficiency.
- **Style**
    - Updated CSS for better alignment and responsiveness of the inbox slide-in and review notice sections.
- **Documentation**
    - No visible changes to end-users.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Crabcyborg commented on classes/helpers/FrmAppHelper.php:
```diff
@@ -3153,6 +3153,7 @@
 			'loading'                       => __( 'Loading&hellip;', 'formidable' ),
 			'nonce'                         => wp_create_nonce( 'frm_ajax' ),
 			'proIncludesSliderJs'           => is_callable( 'FrmProFormsHelper::prepare_custom_currency' ),
+			'inboxSlideIn'                  => FrmInbox::get_inbox_slide_in_value_for_js(),
```
Yes.

garretlaxton commented:
On some pages such as the views page, I don't see the image displayed. Is this correct? I have cleared my browser cache when I noticed this.
![image](https://github.com/Strategy11/formidable-forms/assets/16290508/d949fe93-7ef1-4247-bcd2-86f1fb37e92a)

I can see it on other pages.
![image](https://github.com/Strategy11/formidable-forms/assets/16290508/43778724-edc4-4484-9912-f5394df209ce)

Crabcyborg commented:
Thanks @garretlaxton!

I'll look into that. Something is off.

Crabcyborg commented:
@garretlaxton It looks like I was missing a dependency.

If you pull the latest version of the Pro branch (https://github.com/Strategy11/formidable-pro/pull/4872) it should work now.

garretlaxton reviewed (approved):
Nice, this looks and functions great!

stephywells reviewed (approved):
Awesome! 

I didn't check the review request to see if it matches this new option.

Also, it would be a big help if you could update this form and json view with the expected output for this new option. It would probably be best if the url is a test field so it can come from an S3 url.
https://formidableforms.com/wp-admin/term.php?taxonomy=frm_application&tag_ID=20320

Thanks so much!!

stephywells commented:
Oh! It looks like it would be really helpful if the border radius could be added to the img css so we don't have to add it to the image.

Crabcyborg commented:
Thanks Steph!

> It looks like it would be really helpful if the border radius could be added to the img css so we don't have to add it to the image.

Thanks! I added a border radius style now.

I tested with a placeholder image with no rounded corners,
<img width="438" alt="Screen Shot 2024-03-22 at 9 47 10 AM" src="https://github.com/Strategy11/formidable-forms/assets/9134515/0b7d4c57-0134-4e68-a793-8ed9c6d57955">

> I didn't check the review request to see if it matches this new option.

I requested review before you mentioned that.

I mad a new PR branching off of this one with some review request changes. I tried to drop the WordPress classes and use our Formidable card styling, plus some other classes I've added to try making this a bit easier to share styles.

stephywells reviewed (approved):
Thank you! This looks great.
